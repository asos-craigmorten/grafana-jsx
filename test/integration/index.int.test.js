/** @jsx createObject */
import {
  Annotation,
  Annotations,
  Dashboard,
  Panels,
  Panel,
  Templates,
  Template,
  createObject,
  Row,
} from "../../src";

const requiredDashboardProps = {
  title: "test-title",
};

describe("Integration", () => {
  describe("when providing no configuration", () => {
    it("should create a default dashboard", () => {
      expect(<Dashboard {...requiredDashboardProps} />).toMatchSnapshot();
    });
  });

  describe("when providing custom panels", () => {
    it("should create a dashboard with panels", () => {
      expect(
        <Dashboard {...requiredDashboardProps}>
          <Panels>
            <Panel type={"text"} x={0} y={0} />
            <Panel type={"text"} x={0} y={9} />
          </Panels>
        </Dashboard>
      ).toMatchSnapshot();
    });

    describe("and with custom rows", () => {
      describe("which are collapsed", () => {
        it("should create a dashboard with collapsed row panels", () => {
          expect(
            <Dashboard {...requiredDashboardProps}>
              <Panels>
                <Panel height={1} type={"text"} width={24} x={0} y={0} />
                <Row collapsed={true} title={"test-row-title"} y={1}>
                  <Panel height={1} type={"text"} width={24} x={0} y={1} />
                  <Panel height={1} type={"text"} width={24} x={0} y={2} />
                </Row>
                <Panel height={1} type={"text"} width={24} x={0} y={3} />
              </Panels>
            </Dashboard>
          ).toMatchSnapshot();
        });
      });

      describe("which are not collapsed", () => {
        it("should create a dashboard with uncollapsed row panels", () => {
          expect(
            <Dashboard {...requiredDashboardProps}>
              <Panels>
                <Panel height={1} type={"text"} width={24} x={0} y={0} />
                <Row collapsed={false} title={"test-row-title"} y={1}>
                  <Panel height={1} type={"text"} width={24} x={0} y={1} />
                  <Panel height={1} type={"text"} width={24} x={0} y={2} />
                </Row>
                <Panel height={1} type={"text"} width={24} x={0} y={3} />
              </Panels>
            </Dashboard>
          ).toMatchSnapshot();
        });
      });
    });
  });

  describe("when providing custom templating", () => {
    it("should create a dashboard with templates", () => {
      expect(
        <Dashboard {...requiredDashboardProps}>
          <Templates>
            <Template />
            <Template />
          </Templates>
        </Dashboard>
      ).toMatchSnapshot();
    });
  });

  describe("when providing custom annotations", () => {
    it("should create a dashboard with annotations", () => {
      expect(
        <Dashboard {...requiredDashboardProps}>
          <Annotations>
            <Annotation
              builtIn={0}
              datasource={"test-datasource"}
              name={"test-name"}
            />
          </Annotations>
        </Dashboard>
      ).toMatchSnapshot();
    });
  });
});
