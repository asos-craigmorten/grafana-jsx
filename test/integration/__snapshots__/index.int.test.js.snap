// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration when providing custom annotations should create a dashboard with annotations 1`] = `
Object {
  "annotations": Object {
    "list": Array [
      Object {
        "builtIn": 0,
        "datasource": "test-datasource",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "test-name",
        "type": "dashboard",
      },
    ],
  },
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "links": Array [],
  "panels": Array [],
  "style": "dark",
  "tags": Array [],
  "templating": Object {
    "enable": true,
    "list": Array [],
  },
  "time": Object {
    "from": "now-6h",
    "to": "now",
  },
  "timepicker": Object {
    "collapse": false,
    "enable": true,
    "notice": false,
    "now": true,
    "refresh_intervals": Array [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d",
    ],
    "status": "Stable",
    "time_options": Array [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d",
    ],
    "type": "timepicker",
  },
  "timezone": "browser",
  "title": "test-title",
}
`;

exports[`Integration when providing custom panels and with custom rows which are collapsed should create a dashboard with collapsed row panels 1`] = `
Object {
  "annotations": Object {
    "list": Array [
      Object {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard",
      },
    ],
  },
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "links": Array [],
  "panels": Array [
    Object {
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 0,
      },
      "type": "text",
    },
    Object {
      "collapsed": true,
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 1,
      },
      "panels": Array [
        Object {
          "gridPos": Object {
            "h": 1,
            "w": 24,
            "x": 0,
            "y": 1,
          },
          "type": "text",
        },
        Object {
          "gridPos": Object {
            "h": 1,
            "w": 24,
            "x": 0,
            "y": 2,
          },
          "type": "text",
        },
      ],
      "title": "test-row-title",
      "type": "row",
    },
    Object {
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 3,
      },
      "type": "text",
    },
  ],
  "style": "dark",
  "tags": Array [],
  "templating": Object {
    "enable": true,
    "list": Array [],
  },
  "time": Object {
    "from": "now-6h",
    "to": "now",
  },
  "timepicker": Object {
    "collapse": false,
    "enable": true,
    "notice": false,
    "now": true,
    "refresh_intervals": Array [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d",
    ],
    "status": "Stable",
    "time_options": Array [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d",
    ],
    "type": "timepicker",
  },
  "timezone": "browser",
  "title": "test-title",
}
`;

exports[`Integration when providing custom panels and with custom rows which are not collapsed should create a dashboard with uncollapsed row panels 1`] = `
Object {
  "annotations": Object {
    "list": Array [
      Object {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard",
      },
    ],
  },
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "links": Array [],
  "panels": Array [
    Object {
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 0,
      },
      "type": "text",
    },
    Object {
      "collapsed": false,
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 1,
      },
      "panels": Array [],
      "title": "test-row-title",
      "type": "row",
    },
    Object {
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 1,
      },
      "type": "text",
    },
    Object {
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 2,
      },
      "type": "text",
    },
    Object {
      "gridPos": Object {
        "h": 1,
        "w": 24,
        "x": 0,
        "y": 3,
      },
      "type": "text",
    },
  ],
  "style": "dark",
  "tags": Array [],
  "templating": Object {
    "enable": true,
    "list": Array [],
  },
  "time": Object {
    "from": "now-6h",
    "to": "now",
  },
  "timepicker": Object {
    "collapse": false,
    "enable": true,
    "notice": false,
    "now": true,
    "refresh_intervals": Array [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d",
    ],
    "status": "Stable",
    "time_options": Array [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d",
    ],
    "type": "timepicker",
  },
  "timezone": "browser",
  "title": "test-title",
}
`;

exports[`Integration when providing custom panels should create a dashboard with panels 1`] = `
Object {
  "annotations": Object {
    "list": Array [
      Object {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard",
      },
    ],
  },
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "links": Array [],
  "panels": Array [
    Object {
      "gridPos": Object {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 0,
      },
      "type": "text",
    },
    Object {
      "gridPos": Object {
        "h": 9,
        "w": 12,
        "x": 0,
        "y": 9,
      },
      "type": "text",
    },
  ],
  "style": "dark",
  "tags": Array [],
  "templating": Object {
    "enable": true,
    "list": Array [],
  },
  "time": Object {
    "from": "now-6h",
    "to": "now",
  },
  "timepicker": Object {
    "collapse": false,
    "enable": true,
    "notice": false,
    "now": true,
    "refresh_intervals": Array [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d",
    ],
    "status": "Stable",
    "time_options": Array [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d",
    ],
    "type": "timepicker",
  },
  "timezone": "browser",
  "title": "test-title",
}
`;

exports[`Integration when providing custom templating should create a dashboard with templates 1`] = `
Object {
  "annotations": Object {
    "list": Array [
      Object {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard",
      },
    ],
  },
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "links": Array [],
  "panels": Array [],
  "style": "dark",
  "tags": Array [],
  "templating": Object {
    "enable": true,
    "list": Array [
      Object {
        "allFormat": null,
        "allValue": null,
        "current": Object {},
        "datasource": null,
        "definition": "",
        "hide": 0,
        "includeAll": false,
        "index": -1,
        "label": "",
        "multi": false,
        "name": "",
        "options": Array [],
        "query": "",
        "refresh": 2,
        "regex": "",
        "skipUrlSync": false,
        "sort": 1,
        "tagValuesQuery": "",
        "tags": Array [],
        "tagsQuery": "",
        "type": null,
        "useTags": false,
      },
      Object {
        "allFormat": null,
        "allValue": null,
        "current": Object {},
        "datasource": null,
        "definition": "",
        "hide": 0,
        "includeAll": false,
        "index": -1,
        "label": "",
        "multi": false,
        "name": "",
        "options": Array [],
        "query": "",
        "refresh": 2,
        "regex": "",
        "skipUrlSync": false,
        "sort": 1,
        "tagValuesQuery": "",
        "tags": Array [],
        "tagsQuery": "",
        "type": null,
        "useTags": false,
      },
    ],
  },
  "time": Object {
    "from": "now-6h",
    "to": "now",
  },
  "timepicker": Object {
    "collapse": false,
    "enable": true,
    "notice": false,
    "now": true,
    "refresh_intervals": Array [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d",
    ],
    "status": "Stable",
    "time_options": Array [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d",
    ],
    "type": "timepicker",
  },
  "timezone": "browser",
  "title": "test-title",
}
`;

exports[`Integration when providing no configuration should create a default dashboard 1`] = `
Object {
  "annotations": Object {
    "list": Array [
      Object {
        "builtIn": 1,
        "datasource": "-- Grafana --",
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard",
      },
    ],
  },
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "links": Array [],
  "panels": Array [],
  "style": "dark",
  "tags": Array [],
  "templating": Object {
    "enable": true,
    "list": Array [],
  },
  "time": Object {
    "from": "now-6h",
    "to": "now",
  },
  "timepicker": Object {
    "collapse": false,
    "enable": true,
    "notice": false,
    "now": true,
    "refresh_intervals": Array [
      "5s",
      "10s",
      "30s",
      "1m",
      "5m",
      "15m",
      "30m",
      "1h",
      "2h",
      "1d",
    ],
    "status": "Stable",
    "time_options": Array [
      "5m",
      "15m",
      "1h",
      "6h",
      "12h",
      "24h",
      "2d",
      "7d",
      "30d",
    ],
    "type": "timepicker",
  },
  "timezone": "browser",
  "title": "test-title",
}
`;
